src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlInternalConnectionTds.cs

```diff
@@ -1910,11 +1910,8 @@ private void ResolveExtendedServerName(ServerInfo serverInfo, bool aliasLookup,
                             this,
                             ignoreSniOpenTimeout,
                             timeout.LegacyTimerExpire,
-                            ConnectionOptions.Encrypt,
-                            ConnectionOptions.TrustServerCertificate,
-                            ConnectionOptions.IntegratedSecurity,
+                            ConnectionOptions,
                             withFailover,
-                            ConnectionOptions.Authentication,
                             ConnectionOptions.HostNameInCertificate);
```

Comment:
As long as we are collapsing these parameters into ConnectionOptions, we should include HostNameInCertificate, too.